workflow-core/src/commonMain/kotlin/com/squareup/workflow1/BaseRenderContext.kt

```diff
@@ -173,6 +173,31 @@ public interface BaseRenderContext<out PropsT, StateT, in OutputT> {
    * as a debugging aid
    * @param update Function that defines the workflow update.
    */
+  @Suppress("DEPRECATION")
+  public fun eventHandler(
+    name: String,
+    // Type variance issue: https://github.com/square/workflow-kotlin/issues/891
+    update: WorkflowAction<
+      @UnsafeVariance PropsT,
+      StateT,
+      @UnsafeVariance OutputT
+      >.Updater.() -> Unit
+  ): () -> Unit = eventHandler({ name }, update)
+
+  @Deprecated(
+    "Always provide a debugging name",
+    ReplaceWith("eventHandler(\"TODO: debugging name\", onFailedCast, update)")
+  )
+  public fun eventHandler(
+    // Type variance issue: https://github.com/square/workflow-kotlin/issues/891
+    update:
+    WorkflowAction<@UnsafeVariance PropsT, StateT, @UnsafeVariance OutputT>.Updater.() -> Unit
+  ): () -> Unit = eventHandler("eventHandler", update)
```

Comment:
why do we have to introduce these versions without the `name` parameter just to deprecate it? Is this about the way in which the deprecation detection works with the optional parameter?

Reply:
Yes, exactly.